c_src/erlang_pbc.c

```diff
@@ -815,6 +815,104 @@ pbc_binary_to_element(ErlNifEnv * env, int argc, const ERL_NIF_TERM argv[])
     return term;
 }
 
+static ERL_NIF_TERM
+pbc_elements_to_binary(ErlNifEnv * env, int argc, const ERL_NIF_TERM argv[])
+{
+    if (argc != 1) {
+        return enif_make_badarg(env);
+    }
+
+    if (!enif_is_list(env, argv[0])) {
+        return enif_make_badarg(env);
+    }
+    unsigned int in_len = 0;
+
+    if (!enif_get_list_length(env, argv[0], &in_len)) {
+        return enif_make_badarg(env);
+    }
+
+    ERL_NIF_TERM in_list = argv[0];
+    ERL_NIF_TERM out_list = enif_make_list(env, 0);
+    ERL_NIF_TERM cur;
+    struct pbc_element *element;
+
+    while(enif_get_list_cell(env, in_list, &cur, &in_list)) {
+        if (!enif_get_resource(env, cur, PBC_ELEMENT_RESOURCE, (void**)&element)) {
+            return enif_make_badarg(env);
+        }
+
+        int bytes = element_length_in_bytes(element->element);
+
+        ErlNifBinary bin;
+        if (!enif_alloc_binary(bytes+1, &bin)) {
+            return enif_make_badarg(env);
+        }
+
+        bin.data[0] = (uint8_t) element->field;
+        element_to_bytes(bin.data+1, element->element);
+        out_list = enif_make_list_cell(env, enif_make_binary(env, &bin), out_list);
+    }
+
+    ErlNifBinary result;
+
+    enif_term_to_binary(env, out_list, &result);
+
+    return enif_make_binary(env, &result);
+}
+
+static ERL_NIF_TERM
+pbc_binary_to_elements(ErlNifEnv * env, int argc, const ERL_NIF_TERM argv[])
+{
+
+    if (argc != 2) {
+        return enif_make_badarg(env);
+    }
+
```

Comment:
Why not check if `enif_is_binary(env, argv[1])` and return  `return enif_make_badarg(env)` if not?

Reply:
good idea.